data/random-teams.js

```diff
@@ -2220,6 +2220,7 @@ class RandomTeams extends Dex.ModdedDex {
 		let ability0 = this.getAbility(abilities[0]);
 		let ability1 = this.getAbility(abilities[1]);
 		let ability2 = this.getAbility(abilities[2]);
+		ability = ability0.name;
 		if (abilities[1]) {
 			if (abilities[2] && ability1.rating <= ability2.rating && this.randomChance(1, 2)) {
 				[ability1, ability2] = [ability2, ability1];
```

Comment:
I don't think this is right; you need to choose the ability after checking which one has the best rating. See the `randomSet` code:
https://github.com/Zarel/Pokemon-Showdown/blob/054082e0d2c0b346aa40088a9acff7c903a40bb4/data/random-teams.js#L1152-L1164
https://github.com/Zarel/Pokemon-Showdown/blob/054082e0d2c0b346aa40088a9acff7c903a40bb4/data/random-teams.js#L1293-L1295

Reply:
Fixed in 10f2ff5. Thanks.

Reply:
Oh, right, the first hunk I quoted is already correct for doubles; somehow I hadn't seen that.